crates/pallet-bioauth/src/lib.rs

```diff
@@ -526,6 +548,14 @@ pub mod pallet {
             let active_authentications = possibly_expired_authentications
                 .into_iter()
                 .filter(|possibly_expired_authentication| {
+                    if possibly_expired_authentication.expires_at <= current_moment {
```

Comment:
This doesn't look right, I think we should reuse the condition rather than reevaluate it for each authentication. Also, we deliberately avoided sending an event for expiration before to keep the overhead low, maybe we should send just one event with a list of authentications?

Reply:
Reimplemented.